services/order.py

```diff
@@ -0,0 +1,38 @@
+from django.db import transaction
+from django.contrib.auth import get_user_model
+from django.db.models import QuerySet
+from db.models import Order, Ticket
+
+
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: str = None
+) -> None:
+    with transaction.atomic():
+        user = get_user_model().objects.get(username=username)
+        order = Order.objects.create(user=user)
+
+        if date:
+            order.created_at = date
+            order.save()
+
+        for ticket_data in tickets:
```

Comment:
You should not use create method in a loop - consider using bulk create method

Reply:
bulk_create handle exception and don`t create ticket